### ISSUE TITLE
[Component] [cascader] el-cascader goes out of screen

### ISSUE THREAD

**Issue by roelVerdonschot** [element-plus/element-plus#12620]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.47`
- Element Plus Version: `2.3.4`
- Browser / OS: `Chrome`
- Build Tool: `Vite`

### Related Component

- `el-cascader`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCB2YWx1ZSA9IHJlZihbXSlcbmNvbnN0IG9wdGlvbnMgPSByZWYoW1xuICB7XG4gICAgdmFsdWU6ICcxJyxcbiAgICBsYWJlbDogJ0xvcmVtIGlwc3VtIGRvbG9yIHNpdCBhbWV0LCBjb25zZWN0ZXR1ciBhZGlwaXNjaW5nIGVsaXQsJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAgICB7XG4gICAgICAgICAgIHZhbHVlOiAnMicsXG4gICAgICAgICAgIGxhYmVsOiAnc2VkIGRvIGVpdXNtb2QgdGVtcG9yIGluY2lkaWR1bnQgdXQgbGFib3JlIGV0IGRvbG9yZSBtYWduYSBhbGlxdWEuICcsXG4gICAgICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgICAge1xuICAgICAgICAgICAgICAgdmFsdWU6ICczJyxcbiAgICAgICAgICAgXHRcdGxhYmVsOiAnVXQgZW5pbSBhZCBtaW5pbSB2ZW5pYW0sIHF1aXMgbm9zdHJ1ZCBleGVyY2l0YXRpb24gJyxcbiAgICAgICAgICAgICAgIGNoaWxkcmVuOiBbXG4gICAgICAgICAgICAge1xuICAgICAgICAgICAgICAgdmFsdWU6ICc0JyxcbiAgICAgICAgICAgXHRcdGxhYmVsOiAndWxsYW1jbyBsYWJvcmlzIG5pc2kgdXQgYWxpcXVpcCBleCBlYSBjb21tb2RvIGNvbnNlcXVhdC4gRHVpcyBhdXRlICcsXG4gICAgICAgICAgICAgfVxuICAgICAgICAgICBdXG4gICAgICAgICAgICAgfVxuICAgICAgICAgICBdXG4gICAgICB9LFxuXHRcdF1cbiAgfVxuXSlcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gICA8ZWwtY2FzY2FkZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlXCJcbiAgICAgICAgOm9wdGlvbnM9XCJvcHRpb25zXCJcbiAgICAgICAgY2xhc3M9XCJ3LTEwMFwiXG4gICAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

1. Disable horizontal scroll
2. Add el-cascader to view
3. Add 4 options with children
4. Open cascader and select option
5. Its impossible to select the last option if screen is to small

### What is Expected?

Last panel always fully visible

### What is actually happening?

The last panel is out of screen

### Additional comments

_(empty)_

**Comment by roelVerdonschot** [comment]
Add below styling to Element plus playground somehow it keeps getting removed.

```
<style>
  body {
    overflow-x: hidden;
  }
</style>
```

**Comment by chenxch** [comment]
At present, this kind of scenario may only consider adding `overflow-x: auto` to `.el-cascader-panel`.

**Comment by roelVerdonschot** [comment]
Yes, thats also our current temporary solution but it would be great if the panels don't fit on screen the `.el-cascader__dropdown` gets a `right: 5px;` style instead of the current auto.
